mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py

```diff
@@ -0,0 +1,368 @@
+                # filters on metadata
+                where["meta_data_filter"] = get_metadata_filter(
+                    query.where.args[1].value
+                )
+
+            elif query.where.args[0].parts[-1] == "search_query":
+                # extract the search query
+                where["search_query"] = query.where.args[1].value
+            else:
+                raise NotImplementedError(
+                    f"where clause parameter {query.where.args[0].parts[-1]} is not supported, "
+                    f"only 'meta_data_filter' and 'search_query' are supported"
+                )
+
+        else:
+            raise NotImplementedError(
+                f"Unsupported where clause {query.where.op} operator, only '=' and 'and' is supported"
+            )
+
+        return where
+
+    def query_collection(self, query: ASTNode) -> Response:
+        """
+        Run a select query on the ChromaDB database, filter collection (if where clause)
+        and return result. NB Limit will define the number of documents returned, not the number of rows.
+        """
+
+        collection_name = query.from_table.parts[-1]
+        self._collection = self._client.get_collection(collection_name)
+
+        if hasattr(query.targets[0], "op") and query.targets[0].op == "count":
+
+            # if count is used, return the count of the collection
+            if not isinstance(query.targets[0].args[0], Star):
+                # if count is not using '*' argument, raise error
+                return Response(
+                    resp_type=RESPONSE_TYPE.ERROR,
+                    error_message="Count only supports '*' argument",
+                )
+
+            collection_data = self._collection.count()
+            result = pd.DataFrame(columns=["count"], data=[[collection_data]])
+
+        else:
+
+            # if there is a where clause, parse it to Chromadb query syntax
+            where = self.filter_query(query) if query.where else {}
+
+            if where.get("search_query"):
+                # if 'search_query' provided, run similarity search
+
+                collection_data = self.similarity_search(
+                    filter=where.get(
+                        "meta_data_filter"
+                    ),  # filters on metadata, for now only column name
+                    query=where.get("search_query"),
+                    k=query.limit.value if query.limit else 5,
+                )
+                result = documents_to_df(collection_data)
+
+            else:
+                # if no 'search_query' provided, run a regular query and filter on metadata if provided
+                collection_data = self._collection.get(
+                    where=where.get("meta_data_filter")
+                )
+                result = pd.DataFrame(collection_data)[
+                    ["ids", "documents", "metadatas"]
+                ]
+
+        if query.limit:
```

Comment:
Nitpick: given that `query.limit` affects the value of `k` if it exists, shouldn't this `if` go inside the `else` clause above?